backend/src/neuroagent/tools/context_analyzer_tool.py

```diff
@@ -0,0 +1,365 @@
+"""From url gives back what is on the current page."""
+
+import json
+from pathlib import Path
+from typing import ClassVar
+from urllib.parse import parse_qs, urlparse
+
+from pydantic import BaseModel
+
+from neuroagent.tools.autogenerated_types.entitycore import BrainRegionRead
+from neuroagent.tools.base_tool import BaseTool, EntitycoreMetadata
+
+
+class ContextAnalyzerInput(BaseModel):
+    """Input class for the Context Analyzer tool, empty since no inputs."""
+
+    pass
+
+
+class ContextAnalyzerMetdata(EntitycoreMetadata):
+    """Metadata for the Context Analyzer tool."""
+
+    frontend_url: str
+
+
+class ContextAnalyzerOutput(BaseModel):
+    """Output of the Context Analyzer tool."""
+
+    is_in_project: bool
+    full_page_path: str
+    page_description: str
+
+
+class ContextAnalyzerTool(BaseTool):
+    """Class for the context analyzer tool."""
+
+    name: ClassVar[str] = "context-analyzer-tool"
+    name_frontend: ClassVar[str] = "Context Analyzer"
+    description: ClassVar[
+        str
+    ] = """Gets a description of the current page the user is on. Call this tool when the user needs guidance on the platform.
+    If the user has a vague question about the website USE THIS TOOL."""
+    description_frontend: ClassVar[str] = (
+        """Allows to get the current page the user is navigating. This allows the Agent to help the user navigate the website."""
+    )
+    metadata: ContextAnalyzerMetdata
+    input_schema: ContextAnalyzerInput
+
+    async def arun(self) -> ContextAnalyzerOutput:
+        """From the current url, gives information on the current page.
+
+        Returns
+        -------
+            Description of the current page the user is on, formatted as a string.
+        """
+        if not self.metadata.frontend_url:
+            raise ValueError("Please provide the current frontend url.")
+
+        url = self.metadata.frontend_url
+
+        # retreive description
+        with (Path(__file__).parent.parent / "platform_description.json").open() as f:
+            descriptions = json.load(f)
+
+        # Get url and query params
+        parsed_url = urlparse(url)
+        query_params = parse_qs(parsed_url.query)
+
+        # Remove the base path and split into components
+        path: str = parsed_url.path
+        if not path.startswith("/app/virtual-lab/"):
+            raise ValueError("Invalid Virtual Lab URL")
```

Comment:
This does not work if the path does not start with a `/`.

Unfortunately, I instructed Fabien in the upstream ticket to write `app/...`

Can you maybe make sure it also works in that case (while still supporting `/app..`